addons/mail/static/src/models/notification_group/notification_group.js

```diff
@@ -51,9 +52,12 @@ function factory(dependencies) {
          * @returns {mail.thread|undefined}
          */
         _computeThread() {
-            if (this.res_id) {
+            const notificationsThreadIds = this.notifications.map(notification => notification.message.originThread.id);
+            const threadIds = new Set(notificationsThreadIds);
+            const res_id = (threadIds.size == 1) ? notificationsThreadIds[0] : false;
```

Comment:
camelCase for variables in js, but it can be removed.

If size is different than 1, there should be a quick return with `unlink()` instead.
Then the insert below can directly use `notificationsThreadIds[0]`.

(Also note: equals should always be done with triple operator)